### PR TITLE
DCAR: YouTubeAtom for apps

### PR DESCRIPTION
**Pull request by arelra**
## What does this change?

Enable YoutubeAtom for apps.

As a first step apps will have ads disabled by default.

Hence this PR also refactors the YoutubeAtom into two more directly configurable experiences:
- an ads disabled experience i.e. the base player via `YT.player`
- an ads enabled experienced i.e. the IMA player via `YT.createPlayerForPublishers`

When merged this change will be functionally equivalent as video support has not been activated in Apps yet.

The flow is as follows:

```mermaid
flowchart TD
    classDef yred color:#f00
    A{YoutubeBlockComponent\nenableAds} -->|yes| B{consent and ad-targeting?}
    A{YoutubeBlockComponent\nenableAds} -->|no| C(Disable ads, Unconsented)
    B --> |yes| D(<i class="fa-brands fa-youtube"></i> Player with IMA ads)
    B --> |no| C
    C --> E(<i class="fa-brands fa-youtube"></i> Player with no ads)
    D:::yred
    E:::yred
```

### CODE REVIEW

File: dotcom-rendering/src/components/YoutubeAtom/YoutubeAtom.stories.tsx [hunk 1]
```diff
@@ -1,7 +1,12 @@
 import { css } from '@emotion/react';
+import type { ConsentState } from '@guardian/libs';
```
Comment (arelra) [inline dotcom-rendering/src/components/YoutubeAtom/YoutubeAtom.stories.tsx:2]: The changes to `YoutubeAtom.stories.tsx` are mostly a refactor to make the ads / non ads configuration clearer.

File: dotcom-rendering/src/components/YoutubeAtom/YoutubeAtomPlayer.tsx [hunk 2]
```diff
@@ -349,21 +344,39 @@ const createImaManagerListeners = (uniqueId: string) => {
 	};
 };
 
+const isSignedIn = async (): Promise<boolean> => {
```
Comment (arelra) [inline dotcom-rendering/src/components/YoutubeAtom/YoutubeAtomPlayer.tsx:347]: This is only used for the ads enabled experience so moved from a custom hook to an promisified function. This enables it to be called conditionally whereas that would not be possible with a hook, or without modifying the hook itself.

File: dotcom-rendering/src/components/YoutubeBlockComponent.importable.tsx [hunk 3]
```diff
@@ -223,9 +239,8 @@ export const YoutubeBlockComponent = ({
 				eventEmitters={renderingTarget === 'Web' ? [ophanTracking] : []}
 				format={format}
 				origin={process.env.NODE_ENV === 'development' ? '' : origin}
-				shouldStick={stickyVideos}
+				shouldStick={renderingTarget === 'Web' ? stickyVideos : false}
```
Comment (arelra) [inline dotcom-rendering/src/components/YoutubeBlockComponent.importable.tsx:242]: Disable sticky videos for apps as they visually pass behind ads which are still natively rendered.

File: dotcom-rendering/src/components/YoutubeAtom/YoutubeAtomPlayer.tsx [hunk 4]
```diff
@@ -349,21 +344,39 @@ const createImaManagerListeners = (uniqueId: string) => {
...
+	try {
+		const authStatus = await getAuthStatus();
+		return authStatus.kind == 'SignedInWithCookies' ||
+			authStatus.kind === 'SignedInWithOkta'
+			? true
+			: false;
```
Comment (domlander) [inline dotcom-rendering/src/components/YoutubeAtom/YoutubeAtomPlayer.tsx:353]:
```suggestion
		return authStatus.kind === 'SignedInWithCookies' ||
			authStatus.kind === 'SignedInWithOkta'
```

File: dotcom-rendering/src/components/YoutubeBlockComponent.importable.tsx [hunk 5]
```diff
@@ -193,6 +204,7 @@ export const YoutubeBlockComponent = ({
 		);
 	}
 
+	// TODO video event tracking for apps
```
Comment (domlander) [inline dotcom-rendering/src/components/YoutubeBlockComponent.importable.tsx:207]: Is this TODO for this PR?

File: dotcom-rendering/src/components/YoutubeAtom/YoutubePlayer.ts [hunk 6]
```diff
@@ -95,9 +99,9 @@ class YouTubePlayer {
 			(resolve, reject) => {
 				try {
 					/**
-					 * If enableIma is true, YT.createPlayerForPublishers will be called
-					 * If enableIma is false, the standard new YT.Player constructor will be called
-					 * Listeners are set at expected place for each method
+					 * If enableIma is true, YT.createPlayerForPublishers will be called to initiate IMA ads
+					 * If enableIma is false, the standard YT.Player constructor will be called
+					 * Listeners are set appropriatley for each method
```
Comment (domlander) [inline dotcom-rendering/src/components/YoutubeAtom/YoutubePlayer.ts:104]:
```suggestion
					 * Listeners are set appropriately for each method
```

**Review by domlander — APPROVED** [review]
I haven't tested locally, but I gave the code another look through

File: dotcom-rendering/src/components/YoutubeAtom/YoutubeAtom.stories.tsx [hunk 7]
```diff
@@ -336,99 +364,130 @@ export const PausesOffscreen = {
...
+ * Duplicate YoutubeAtom tests but with ads enabled.
+ *
+ * The YouTube ads integration (IMA) only works on allow listed https domains.
+ * The tests require running:
+ * - DCR as normal on port 3030
+ * - scripts/nginx/setup.sh to setup nginx on https://r.thegulocal.com
+ *
+ * The ad enabled tests are a convenience for manual testing.
+ *
+ */
```
Comment (arelra) [inline dotcom-rendering/src/components/YoutubeAtom/YoutubeAtom.stories.tsx:377]: Disabled snapshots for the ads enabled tests as they require an allow listed https domain

**Pull request merged**